source/fundamentals/crud/read-operations/count.txt

```diff
@@ -51,12 +51,28 @@ Accurate Count
 --------------
 
 To count the number of documents that match your query filter, use the
-``CountDocuments()`` method.
+``CountDocuments()`` method. If you pass an empty query filter, this method
+returns the total number of documents in the collection.
 
 .. tip::
 
-   If you pass an empty query filter, this method returns the total
-   number of documents in the collection.
+   Using ``CountDocuments()`` to return the total number of documents in the
+   collection results in a collection scan. You can avoid a collection scan and
+   improve the performance of this method by using a :manual:`hint
+   </reference/method/cursor.hint>` to take advantage of the built-in index on
+   the ``_id`` field. Use this technique only when calling ``CountDocuments()``
+   with an empty query parameter.
+
+   .. code-block:: go
+      :emphasize-lines: 2, 4
+
+      filter := bson.D{{}}
+      opts := options.Count().SetHint("_id_")
+
+      count, err := coll.CountDocuments(context.TODO(), filter, opts)
```

Comment:
Q: do we need to define `filter` if it's just empty? Instead of:
```suggestion
      opts := options.Count().SetHint("_id_")

      count, err := coll.CountDocuments(context.TODO(), filter, bson.D{{}})
```